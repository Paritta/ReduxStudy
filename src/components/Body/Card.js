import React from "react";
import styled from "styled-components";
import EmptyImage from "../../asset/EmptyImage.png";
import PropTypes from "prop-types";
import ReactLoading from "react-loading";

const propTypes = {
    image: PropTypes.object,
    imageReceiveRequest: PropTypes.func,
    ImageReceive: PropTypes.object,
};

const defaultTypes = {
    image: {},
    imageReceiveRequest() {},
    ImageReceive: {},
};

const Wrapper = styled.div`
    margin: 0;
    padding: 0;
    
    width: 450px;
    border-radius: 10px 10px 0 0;
    overflow: hidden;
`;

const View = styled.div`
    margin: 0;
    padding: 0;

    width: 450px;
    height: 250px;
    
    background: url(${props => props.ImageUrl ? props.ImageUrl : EmptyImage}) center center;
    background-size: ${props => props.ImageUrl ? 100 : 40}px;
    
    display: flex;
    align-items: center;
    justify-content: center;
    
    color: white;
    
    transform: scale(10);
    
    z-index: 5;
`;

const Info =styled.div`
    margin: 0;
    padding: 0;
    
    overflow: hidden;
    width: 450px;
    height: 100px;
    
    background: hsla(0, 72%, 0%, 0.4);
    
    z-index: 10;
    
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
`;

const Title = styled.div`
    font-family: 'Hanna', fantasy;
    font-size: 40px;
    font-weight: 500;
    color: white;
`;

const WrappedReactLoading = styled.div`
    transform: scale(2);
    position: absolute;
    
    top: 15%;
    left: 40%;
    
    padding: 80px 0;
`;

const LocationDiv = styled.div`   
    font-family: 'Hanna', fantasy;
    font-size: 22px;
    font-weight: 400;
    color: white;
`;

export class Card extends React.Component {
    componentDidMount () {
        this.props.imageReceiveRequest(this.props.postId)
    }

    render () {
        const { item, ImageReceive } = this.props;
        const { StudyTitle, Location } = item.data.values;
        const Pending = ImageReceive.pending;
        const ImageData = ImageReceive.data;
        let ImageUrl = false;
        if(this.props.key !== undefined && ImageReceive.data.length !== 0 && ImageReceive.data !== undefined) {
            ImageUrl = ImageReceive.data[this.props.key].Url;
        }
        const PostImageKey = item.data.PostImageKey;

        return (
            <div>
                {/* 이미지 로딩 */}
                {
                    Pending && ImageData.length === 0 &&
                    <WrappedReactLoading>
                        <ReactLoading type="cylon" color="palevioletred"/>
                    </WrappedReactLoading>
                }

                {/* 이미지가 없을 때는 EmptyImage  */}
                {
                    PostImageKey === undefined &&
                    <Wrapper>
                        <View
                            ImageUrl={false}
                        >
                            .
                        </View>
                    </Wrapper>
                }

                {/* 이미지가 있을 때는 ImageUrl, 이미지 등록 버튼 삭제 */}
                {
                    PostImageKey !== undefined &&
                    <Wrapper>
                        <View
                            ImageUrl={ImageUrl}
                        >
                        </View>
                    </Wrapper>
                }
                {/*<Wrapper>*/}
                    {/*<View>*/}
                        {/*<FaCameraRetro/>*/}
                    {/*</View>*/}
                {/*</Wrapper>*/}

                <Info>
                    <Title>
                        { StudyTitle }
                    </Title>
                    <LocationDiv>
                        { Location }
                    </LocationDiv>
                </Info>
            </div>
        );
    }
}

Card.propTypes = propTypes;
Card.defaultTypes = defaultTypes;

export default Card;