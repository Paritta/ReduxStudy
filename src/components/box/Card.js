import React from "react";
import styled from "styled-components";
import oc from "open-color";
import FaCameraRetro from "react-icons/lib/fa/camera-retro";

const propTypes = {
};

const defaultTypes = {
};

const Wrapper = styled.div`
    display: inline-block;
    
    margin: 0;
    padding: 0;
    
    border-radius: 15px;
    overflow: hidden;
`;

const View = styled.div`
    margin: 0;
    padding: 0;

    width: 450px;
    height: 300px;
    
    background: ${oc.gray[1]};
    
    display: flex;
    align-items: center;
    justify-content: center;
    
    color: white;
    
    transform: scale(10);
    
    z-index: 5;
`;

const Info =styled.div`
    margin: 0;
    padding: 0;
    
    overflow: hidden;
    width: 450px;
    height: 100px;
    
    background: pink;
    
    z-index: 10;
`;

const Card = () => {
    return (
        <div>
            <Wrapper>
                <View>
                    <FaCameraRetro/>
                </View>
            </Wrapper>
            <Info/>
        </div>
    );
};

Card.propTypes = propTypes;
Card.defaultTypes = defaultTypes;

export default Card;