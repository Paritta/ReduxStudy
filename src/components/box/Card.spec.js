import React from "react";
import { Card } from "./Card";
import "jest-styled-components";
import ReactTestRenderer from "react-test-renderer";

it("Component render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Card />
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
