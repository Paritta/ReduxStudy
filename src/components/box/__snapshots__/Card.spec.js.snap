// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render correctly 1`] = `
.c0 {
  display: inline-block;
  margin: 0;
  padding: 0;
  border-radius: 15px;
  overflow: hidden;
}

<div
  className="c0"
/>
`;
